src/main/java/liquibase/ext/mongodb/statement/AbstractRunCommandStatement.java

```diff
@@ -84,6 +95,80 @@ protected void checkResponse(final Document responseDocument) throws MongoExcept
         }
     }
 
+    /**
+     * Updates the rows affected count in the Liquibase scope based on MongoDB command response
+     */
+    protected void updateRowsAffected(Document response) {
+        AtomicInteger rowsAffected = Scope.getCurrentScope().get(JdbcExecutor.ROWS_AFFECTED_SCOPE_KEY, AtomicInteger.class);
+        Boolean shouldUpdateRowsAffected = Scope.getCurrentScope().get(JdbcExecutor.SHOULD_UPDATE_ROWS_AFFECTED_SCOPE_KEY, Boolean.class);
+
+        if (rowsAffected != null && Boolean.TRUE.equals(shouldUpdateRowsAffected)) {
+            int count = extractAffectedCount(response);
+            if (count > -1) {
+                rowsAffected.addAndGet(count);
+            }
+        }
+    }
+
+    /**
+     * Extracts the number of affected documents from MongoDB command response
+     */
+    protected int extractAffectedCount(Document response) {
+        // For collection-level operations
+        if (isCollectionOperation()) {
+            double ok = response.get(OK) instanceof Integer ?
+                    (double) response.getInteger(OK) :
+                    response.getDouble(OK);
+            if (ok == 1.0d) {
+                return 1;
+            }
+            return 0;
+        }
+
+        // For insert operations
+        Integer nInserted = response.getInteger(N);
+        if (nInserted != null) {
+            return nInserted;
+        }
+
+        // For update operations
+        Integer nModified = response.getInteger(N_MODIFIED);
+        if (nModified != null) {
+            return nModified;
+        }
+
+        // For delete operations
+        Integer nRemoved = response.getInteger(N_REMOVED);
+        if (nRemoved != null) {
+            return nRemoved;
+        }
+
+        // For generic operations
+        Integer n = response.getInteger(N);
+        if (n != null) {
+            return n;
+        }
+
+        // Default successful operation count
+        double ok = response.get(OK) instanceof Integer ?
```

Comment:
this last resort is doing the same as `if (isCollectionOperation()) {..}` part. Do we need both?

Reply:
you're right. Fixed.